src/Reflection/ReflectionClass.php

```diff
@@ -268,11 +301,11 @@ public function getProperties()
                 // We also skip final, to improve error handling and consistent behaviour,
                 // otherwise final private $foo would be parsed and private final $bar
                 // would not be parsed.
-                $var_loc = $this->tokens->next($vis_loc, [T_COMMENT, T_WHITESPACE, T_STATIC, T_FINAL]);
-                if ($this->tokens->type($var_loc) === T_VARIABLE) {
+                $var_loc = $this->getTokenStream()->next($vis_loc, [T_COMMENT, T_WHITESPACE, T_STATIC, T_FINAL]);
+                if ($this->getTokenStream()->type($var_loc) === T_VARIABLE) {
```

Comment:
Not sure if this change makes me very happy, it's quite some overhead to call a function each time. If you really wish to keep this method as is, would it be possible to assign the token stream to a local variable ?

Same goes for all the other functions.

Reply:
Line 589~591:

```php
    private function getTokenStream()
    {
        if (! $this->tokens) {
            $this->tokens = new TokenStream(file_get_contents($this->filename));
        }

        return $this->tokens;
    }
```

Reply:
I know it caches the tokens internally, but calling  `getTokenStream` every time is a considerable amount of overhead that should be avoided imo.

```php
$tokens = $this->getTokenStream();

$token->...;
```